examples/reverse.rs

```diff
@@ -11,8 +11,13 @@ fn reverse<T: Clone>(xs: &[T]) -> Vec<T> {
 }
 
 fn main() {
-    fn equality_after_applying_twice(xs: Vec<isize>) -> bool {
+    fn equality_after_applying_twice_vec(xs: Vec<isize>) -> bool {
         xs == reverse(&reverse(&xs))
     }
-    quickcheck(equality_after_applying_twice as fn(Vec<isize>) -> bool);
+    quickcheck(equality_after_applying_twice_vec as fn(Vec<isize>) -> bool);
+
+    fn equality_after_applying_twice_array(xs: [u8; 10]) -> bool {
+        &xs[..] == &reverse(&reverse(&xs))[..]
+    }
+    quickcheck(equality_after_applying_twice_array as fn([u8; 10]) -> bool);
```

Comment:
@hauleth Could you remove the arrays from this example? I'd like to keep this example short and sweet.